packages/contracts/contracts/CyberMarketplace.sol

```diff
@@ -0,0 +1,125 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.9;
+
+import '@openzeppelin/contracts/utils/Counters.sol';
+import '@openzeppelin/contracts/utils/Address.sol';
+//import '@openzeppelin/contracts/access/Ownable.sol';
+import '@openzeppelin/contracts/token/ERC1155/IERC1155.sol';
+import '@openzeppelin/contracts/utils/introspection/ERC165Checker.sol';
+import '@openzeppelin/contracts/token/ERC1155/utils/ERC1155Holder.sol';
+
+bytes4 constant NFT_SIG = 0xd9b67a26;
```

Comment:
you can properly call the interface in place of the encoded value

Reply:
I used ER165 to first check if the contract is a valid ERC1155, thought it provides better error message.

```solidity
 require(
      ERC165Checker.supportsInterface(nftContract, NFT_SIG),
      'Invalid NFT'
    );
```

I guess if we're going to whitelist only certain contracts, then we dont need it

Reply:
No it's perfect, I'm talking about NFT_SIG only, the rest is correct.
But move this logic into an init function to setup one time the contracts that are allowed. It's good to keep this check in case we try to whitelist wrong contracts address

Reply:
bonus function to add new contract to the whitelist 😉

Reply:
something like `bytes4 constant NFT_SIG = type(IERC1155).interfaceId;`

Reply:
Perfect, don't need to store it you can dynamically use the `type(IERC1155).interfaceId`